bridges/dw_hdmi/aw_de2_dw_hdmi.c

```diff
@@ -912,6 +1198,60 @@ MODULE_DEPEND(aw_de2_dw_hdmi, aw_de2_hdmi_phy, 1, 1, 1);
 
 /* Rockchip */
 
+static void
+rk_hdmi_configure(struct rk_dw_hdmi_softc *sc)
+{
+	uint32_t reg;
+
+	/* Select VOP Little for HDMI. */
+	reg = SYSCON_READ_4(sc->grf, GRF_SOC_CON20);
+	reg &= ~CON20_HDMI_VOP_SEL_M;
+	reg |= CON20_HDMI_VOP_SEL_L;
+	SYSCON_WRITE_4(sc->grf, GRF_SOC_CON20, reg);
+}
+
+static int
+rk_hdmi_clk_enable(device_t dev)
+{
+	struct rk_dw_hdmi_softc *sc;
+	uint64_t rate;
+	int error;
+	int i;
+
+	sc = device_get_softc(dev);
+
+	for (i = 0; i < RK_CLK_NENTRIES; i++) {
```

Comment:
The common clock will be enabled twice as we enable the required ones in dw_hdmi.

Reply:
Yes, but the rest of RK clocks have to be enabled *after* common clocks and *before* dw_attach(), so I have to enable them twice.